### PR TITLE
feat: add new check to prevent TS direct dep

### PR DESCRIPTION
**Pull request by WillieRuemmele**
@W-15981337@

see: 
https://github.com/forcedotcom/source-deploy-retrieve/pull/1373

first commit, without TS dep, passes https://github.com/forcedotcom/source-deploy-retrieve/actions/runs/10045614955/job/27763259371

second commit, with TS dep, fails https://github.com/forcedotcom/source-deploy-retrieve/actions/runs/10045668116/job/27763423256

note: will need to change `@wr/noTS` reference to `@main` before merging

### CODE REVIEW

File: .github/workflows/unitTest.yml
Comment (mshanemc) [inline .github/workflows/unitTest.yml:1]: most repos don't call this.  They call the linux/windows directly.

https://github.com/salesforcecli/plugin-user/blob/844a7cd0bcc1a3698f68a4ee3e9836d5ff5f9b97/.github/workflows/test.yml#L11

Comment (mshanemc) [reply .github/workflows/unitTest.yml:1]: so we're either going to have to put it in every repo OR tuck it into something weird (ex: lockfile check or linux ut etc) that's already everywhere

Comment (WillieRuemmele) [reply .github/workflows/unitTest.yml:1]: the `devDeps` are required for UTs to run though

**Review by mshanemc — APPROVED** [review]
don't forget to change the reference back to main

**Pull request merged**